src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private static final double INCREASE_ARRAY = 1.5;
+    private Object[] arraylist = new Object[ARRAY_SIZE];
+    private int size;
+
+    public ArrayList() {
+        this.arraylist = arraylist;
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 <= arraylist.length) {
+            arraylist[size] = value;
+        } else if (size + 1 > ARRAY_SIZE) {
+            grow();
+            arraylist[size] = value;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        if (size + 1 == arraylist.length) {
```

Comment:
```suggestion
        if (size == arraylist.length) {
```
if arraylist.length is 10 and size is 9 - we still can add one element

Reply:
Done.